technic/machines/switching_station.lua

```diff
@@ -135,36 +141,38 @@ local function add_cable_node(nodes, pos, network_id, queue)
 end
 
 -- Generic function to add found connected nodes to the right classification array
-local check_node_subp = function(PR_nodes, RE_nodes, BA_nodes, SP_nodes, all_nodes, pos, machines, tier, sw_pos, from_below, network_id, queue)
+local function check_node_subp(PR_nodes, RE_nodes, BA_nodes, SP_nodes, all_nodes, pos, machines, tier, sw_pos, from_below, network_id, queue)
 
 	local distance_to_switch = vector.distance(pos, sw_pos)
 	if distance_to_switch > switch_max_range then
 		-- max range exceeded
-		return
+		return true
 	end
 
 	technic.get_or_load_node(pos)
 	local name = minetest.get_node(pos).name
 
 	if technic.is_tier_cable(name, tier) then
-		add_cable_node(all_nodes, pos,network_id, queue)
+		add_cable_node(all_nodes, pos, network_id, queue)
 	elseif machines[name] then
 		--dprint(name.." is a "..machines[name])
 
 		local meta = minetest.get_meta(pos)
-		local net_id_key = tier.."_network"
+		local pos_hash = minetest.hash_node_position(pos)
 		local net_id_new = network_id
-		local net_id_old = meta:get_string(net_id_key)
-		if net_id_old == "" then
-			meta:set_string(net_id_key, net_id_new)
+		local net_id_old = technic.cables[pos_hash]
+		if net_id_old == nil then
+			technic.cables[pos_hash] = net_id_new
 		elseif net_id_old ~= net_id_new then
 			-- do not allow running pos from multiple networks, also disable switch
 			overloaded_networks[net_id_old] = minetest.get_us_time() + (overload_reset_time * 1000 * 1000)
 			overloaded_networks[net_id_new] = overloaded_networks[net_id_old]
 			-- delete caches for conflicting network
 			technic.networks[net_id_old] = nil
-			meta:set_string(net_id_key, net_id_new)
+			technic.networks[net_id_new] = nil
```

Comment:
```
technic.networks[net_id_old] = nil
technic.networks[net_id_new] = nil
```
These actually should not be cleared here, also if early return fires back `technic.networks[net_id_new] = nil` cannot be here as it is network that is currently processed.

Function `reset_overloaded(network_id)` already clears these after restart timer has expired for network in question.
Both could still be used to clean up cables in a more controlled way (`technic.networks[network_id].all_nodes`).